apps/cowswap-frontend/src/common/hooks/useApproveCurrency.ts

```diff
@@ -0,0 +1,22 @@
+import { useCallback } from 'react'
+
+import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
+
+import { useShouldZeroApprove, useZeroApprove } from 'modules/zeroApproval'
+
+import { useTradeApproveCallback } from '../containers/TradeApprove'
+
+// TODO: allocate a module for all approval process logic
+export function useApproveCurrency(amountToApprove: CurrencyAmount<Currency> | undefined) {
```

Comment:
Extracted this hook from `TradeApproveButton` and reused in `useOrdersTableTokenApprove`